### DISCUSSION TITLE
Numaflow UDF Java SDK plans?

### DISCUSSION THREAD

**Discussion by lalitmathwani** [numaproj/numaflow#247] [Q&A] [answered]
Hello!

Thanks a lot for the numaflow contribution to the community ! 

At Adobe, we are in the process of rolling out an Argo based solution for a new set of Platform services.  We were very interested in experimenting with Numaflow as an alternative to Argo Event/Workflows for one of our  workflows that we expect will see fairly heavy use. 

Had a quick question about UDF SDKs - are there plans to expand beyond Go and Python ? We were especially interested in a Java SDK - if that's in the works :) 

Cheers!

**Comment by vigith** [comment, marked as answer]
Yes, we do have plans for Java SDK, and even SDKs in other languages. It is just that we haven't prioritized it. Would you be interested in contributing?

**Comment by lalitmathwani** [comment]
@vigith  Thanks a lot ! That's great to hear ..
Re contributing , would love to - but can't promise anything. Let me try to get a better understanding of what's involved in creating a SDK in the next week or so. Cheers!

**Reply by dseapy** [reply]
i've started trying out numaflow recently & come from a scala background, so also just started working to throw an java sdk together to make use of some java & scala libraries I make use of in functions & sinks (https://github.com/dseapy/numaflow-java).

still a wip, may have a thing running in the next day or two :crossed_fingers: .  feel free to use whatever bits might be useful.  also happy to collab on an official java sdk.

**Reply by lalitmathwani** [reply]
Thanks a ton @dseapy. Definitely will love to collab on the java SDK if we decide to pursue the numaflow option further.

**Comment by vigith** [comment]
thanks to @dseapy, we have Java SDK for Numaflow, https://github.com/numaproj/numaflow-java

**Comment by whynowy** [comment]
Close it as the java sdk is available.
